cartographer/mapping/map_builder.cc

```diff
@@ -99,6 +99,30 @@ int MapBuilder::AddTrajectoryBuilder(
   return trajectory_id;
 }
 
+int MapBuilder::AddTrajectoryBuilder(
```

Comment:
Remove the original AddTrajectoryBuilder and take the (to be created) TrajectoryBuilderOptions proto here. `CHECK` that the correct oneof is set here (the one that matches the settings in the original `MapBuilderOptions` and therefore the sparse pose graph).